### ISSUE TITLE
Saving of Intermediate Results

### ISSUE THREAD

**Issue by FFroehlich** [ICB-DCM/pyPESTO#29]
I think the best way to achieve this is to store information such as n_feval, time, parameters every time a lower objective function value is achieved. 

This would mean that we have to store information such as n_feval, n_jeval etc manually, but that is probably what we want anyways to make computation of these values more consistent across different optimization routines. Alternative would be to use a callback function but this may not be consistently possible for all optimizers

This may not work well in multi-threaded setting, but I think embarassingly parallal parallelization makes more sense for our applications anyways.

**Comment by yannikschaelte** [comment]
i agree that it would be good to track n_feval etc. for comparability reasons, for benchmarking i did that anyways since the optimizers have different notions here. question: do we want to then disregard the output concerning function calls etc. from the used optimizer?

**Comment by FFroehlich** [comment]
Yes, I would disregard the optimizer results.
